### PR TITLE
KOGITO-8015: Serverless Logic Web Tools UX Redesign

### PR DESCRIPTION
**Pull request by fantonangeli**
**Jiras:**  
https://issues.redhat.com/browse/KOGITO-8015
https://issues.redhat.com/browse/KOGITO-9171

**Description**
Serverless Logic Web Tools UX Redesign to align with MAS UX and future integration of Web Tools on Red Hat Managed Service

**Preview**
https://fantonangeli.github.io/kie-tools/kogito-8015/#/

**Special thanks**
@caponetto who guided me and helped a lot, @paulovmr, @bxf12315, @tomasdavidorg and everyone helped and reviewed all the PRs in this epic.

### CODE REVIEW

File: packages/serverless-logic-web-tools/src/homepage/recentModels/WorkspacesTable.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+ */
+import {
+  PromiseStateStatus,
+  PromiseStateWrapper,
+  useLivePromiseState,
+} from "@kie-tools-core/react-hooks/dist/PromiseState";
+import { useWorkspaces, WorkspaceFile } from "@kie-tools-core/workspaces-git-fs/dist/context/WorkspacesContext";
+import { WorkspaceDescriptor } from "@kie-tools-core/workspaces-git-fs/dist/worker/api/WorkspaceDescriptor";
+import { WorkspaceKind } from "@kie-tools-core/workspaces-git-fs/dist/worker/api/WorkspaceOrigin";
+import "@patternfly/react-core/dist/styles/base.css";
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/homepage/recentModels/WorkspacesTable.tsx]: Please remove this import.

File: packages/serverless-logic-web-tools/src/homepage/recentModels/workspaceFiles/WorkspaceFilesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from "react";
+import { WorkspaceFile } from "@kie-tools-core/workspaces-git-fs/dist/context/WorkspacesContext";
+import "@patternfly/react-core/dist/styles/base.css";
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/homepage/recentModels/workspaceFiles/WorkspaceFilesTable.tsx]: Please remove this import.

File: packages/serverless-logic-web-tools/src/homepage/recentModels/WorkspacesTable.tsx [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+export function WorkspacesTable(props: WorkspacesTableProps) {
+  const { workspaceDescriptors, selectedWorkspaceIds, onClearFilters, searchValue, page, perPage } = props;
+  const [activeSortIndex, setActiveSortIndex] = useState<number>(3);
+  const [activeSortDirection, setActiveSortDirection] = useState<"asc" | "desc">("desc");
+  const workspaces = useWorkspaces();
+
+  const [allWorkspacePromises] = useLivePromiseState<WorkspaceFile[][]>(
+    useMemo(
+      () => () => Promise.all(workspaceDescriptors.map((w) => workspaces.getFiles(w))),
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/homepage/recentModels/WorkspacesTable.tsx:86]: Is it necessary? `() => () =>`

File: packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts [hunk 4]
```diff
@@ -13,8 +13,11 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { deleteAllIndexedDBs } from "../support/e2e";
+
 describe("Serverless Logic Web Tools - Recent model test", () => {
   beforeEach(() => {
+    deleteAllIndexedDBs();
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts:20]: Was there any issue with the tests? Cypress should automatically clear the storage before each test. https://docs.cypress.io/api/commands/clearlocalstorage

File: packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts [hunk 5]
```diff
@@ -39,8 +42,10 @@ describe("Serverless Logic Web Tools - Recent model test", () => {
 
     // open again from main page
     cy.ouia({ ouiaId: "app-title" }).click();
-    cy.ouia({ ouiaId: "recent-models-section" })
-      .find("article h3")
+    cy.goToSidebarLink({ ouiaId: "recent-models-nav" });
+
+    cy.ouia({ ouiaId: "OUIA-Generated-TableRow-2" })
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts:47]: Would be safer to just delete the first row? Or is it rely on the `deleteAllIndexedDBs`?

File: packages/serverless-logic-web-tools/src/AppConstants.ts [hunk 6]
```diff
@@ -15,7 +15,8 @@
  */
 
 export const APP_NAME = "Serverless Logic Web Tools";
-
+export const SERVERLESS_LOGIC_WEBTOOLS_DOCUMENTATION_URL =
+  "https://kiegroup.github.io/kogito-docs/serverlessworkflow/latest/tooling/serverless-logic-web-tools/serverless-logic-web-tools-overview.html";
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/AppConstants.ts:19]: Do we have jira for the update of the documentation? The documentation and the screenshots will be out of date when this PR is merged.

File: packages/serverless-logic-web-tools/src/homepage/overView/UploadCard.tsx [hunk 7]
```diff
@@ -1,11 +1,11 @@
 /*
- * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/overView/UploadCard.tsx]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/serverless-logic-web-tools/src/homepage/pageTemplate/OnlineEditorPage.tsx [hunk 8]
```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/pageTemplate/OnlineEditorPage.tsx]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/serverless-logic-web-tools/src/homepage/overView/ImportFromUrlCard.tsx [hunk 9]
```diff
@@ -1,11 +1,11 @@
 /*
- * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/overView/ImportFromUrlCard.tsx]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/serverless-logic-web-tools/src/homepage/overView/NewModelCard.tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/overView/NewModelCard.tsx]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/serverless-logic-web-tools/src/homepage/overView/Overview.tsx [hunk 11]
```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/overView/Overview.tsx]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/serverless-logic-web-tools/src/homepage/routes/HomePageRoutes.tsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/routes/HomePageRoutes.tsx]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/serverless-logic-web-tools/src/homepage/uiNav/HomePageNav.tsx [hunk 13]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/uiNav/HomePageNav.tsx]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/serverless-logic-web-tools/src/homepage/breadcrumb/HomepageBreadcrumb.tsx [hunk 14]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/breadcrumb/HomepageBreadcrumb.tsx]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/serverless-logic-web-tools/src/settings/openshift/OpenShiftSettingsSimpleConfig.tsx [hunk 15]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/settings/openshift/OpenShiftSettingsSimpleConfig.tsx]:
```suggestion
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
```

File: packages/serverless-logic-web-tools/src/homepage/styles.css [hunk 16]
```diff
@@ -0,0 +1,11 @@
+.kogito-tools-common--aboutModalBox .pf-c-about-modal-box__hero {
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/homepage/styles.css]: I think this file is not being used anywhere. Can you please delete it?

File: packages/serverless-logic-web-tools/src/navigation/RoutesSwitch.tsx [hunk 17]
```diff
@@ -15,52 +15,56 @@
...
   const routes = useRoutes();
-  const supportedExtensions = useMemo(() => supportedFileExtensionArray.join("|"), []);
+  const isRouteInSettingsSection = useRouteMatch(routes.settings.home.path({}));
+  const buildInfo = useMemo(() => {
+    return process.env["WEBPACK_REPLACE__buildInfo"];
+  }, []);
+  const pageContainerRef = useRef<HTMLDivElement>(null);
+  const [isNavOpen, setIsNavOpen] = useState(true);
+
+  const renderPage = (routeProps: {
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/navigation/RoutesSwitch.tsx]: Do we need this `routeProps` arg?

File: packages/serverless-logic-web-tools/src/homepage/overView/Overview.tsx [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+          <Title headingLevel="h1" size="2xl">
+            Welcome to Serverless Logic Web Tools
+          </Title>
+          <StackItem>
+            <Text className="appsrv-marketing--banner__tagline pf-u-color-200">
+              Add-on service to create and synchronize your Serverless Workflow, Decision files, and Dashbuilder files
+            </Text>
+            <Text component={TextVariants.p}>
+              The Serverless Logic Web Tools is a web application that enables you to create and synchronize your
+              Serverless Workflow, Decision files, and Dashbuilder files in a single interface. Also, the Serverless
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/overView/Overview.tsx]: Shouldn't be sentence without `files` and add `Serverless` before decision to keep the consistency with the Create menu?
```suggestion
              Serverless Workflow, Serverless Decision, and Dashbuilder files in a single interface. Also, the Serverless
```

File: packages/serverless-logic-web-tools/src/homepage/overView/Overview.tsx [hunk 19]
```diff
@@ -0,0 +1,215 @@
...
+                src="https://www.youtube.com/embed/t3ahWJjiGoU"
+                title="YouTube video player"
+                allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+                allowFullScreen
+              ></iframe>
+            </Card>
+          </GridItem>
+          <GridItem md={5}>
+            <TextContent>
+              <Text className="pf-u-color-200 pf-u-ml-md">In this video, you&apos;ll learn how to:</Text>
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/overView/Overview.tsx]: I think using the `will` is more formal.
```suggestion
              <Text className="pf-u-color-200 pf-u-ml-md">In this video, you will learn how to:</Text>
```

**Review by caponetto — COMMENTED** [review]
Hey @fantonangeli! Well done! The result is amazing!
I've left only a few minor comments inline.

Also, two notes:
1) I see that sample-related files is under `home/samples` folder. Can you please move them to just `samples/` folder in the `src` root?
2) I think we are not using the `kogito_log_workbranch.svg` file. Can you please remote it?

Thanks!

File: packages/serverless-logic-web-tools/src/homepage/overView/Overview.tsx [hunk 20]
```diff
@@ -0,0 +1,215 @@
...
+              <List className="app-services-ui--icon-list">
+                <ListItem>Create a Serverless Workflow, a Dashboard or a Decision.</ListItem>
```
Comment (tomasdavidorg) [inline packages/serverless-logic-web-tools/src/homepage/overView/Overview.tsx]:
```suggestion
                <ListItem>Create a Serverless Workflow, a Dashboard or a Serverless Decision.</ListItem>
```

File: packages/serverless-logic-web-tools/src/homepage/recentModels/WorkspacesTable.tsx (hunk 3 above)
Comment (fantonangeli) [reply packages/serverless-logic-web-tools/src/homepage/recentModels/WorkspacesTable.tsx:86]: It's required by `useLivePromiseState`. Please correct me if I'm wrong.
Signature: 
`useLivePromiseState<T>(
  promiseDelegate: (() => Promise<T>) | { error: string }
): [PromiseState<T>, (canceled: Holder<boolean>) => void]`

Comment (caponetto) [reply packages/serverless-logic-web-tools/src/homepage/recentModels/WorkspacesTable.tsx:86]: Missed that. That's right!

**Review by caponetto — APPROVED** [review]

**Review by tomasdavidorg — CHANGES_REQUESTED** [review]
@fantonangeli @caponetto I finish the testing of this PR I found some issues or uncertainties:
- There is no dialog after deleting a single line from the recent model table. Is this correct?
- Clicking “Need help getting started…” will open the first tutorial step but not the main page of the tutorial. Is it correct?
- Close button will not close the tutorial. Steps to reproduce: Click back until the continue button is visible. Then click the continue button and click close button.
- Deleting one model will not update selected models. (Select all -> Delete one model -> The same number is still marked as selected)
- Is the only way how to get from settings to click the main logo? It might be confusing for the users to get back to main navigation.

**Comment by fantonangeli** [comment]
Thank you @tomasdavidorg for your review.

> @fantonangeli @caponetto I finish the testing of this PR I found some issues or uncertainties:
> 
>     * There is no dialog after deleting a single line from the recent model table. Is this correct?

I will change this.



>     * Clicking “Need help getting started…” will open the first tutorial step but not the main page of the tutorial. Is it correct?

I made the QuickStart start from the specific section according to the page to lets the user focus directly on what he needs. 
If we want I can change this behaviour starting always from the first page of the tutorial.




>     * Close button will not close the tutorial. Steps to reproduce: Click back until the continue button is visible. Then click the continue button and click close button.

Same issue here: https://quickstarts.netlify.app/
Opened issue: https://github.com/patternfly/patternfly-quickstarts/issues/249
 
 
 
 
>     * Deleting one model will not update selected models. (Select all -> Delete one model -> The same number is still marked as selected)

This should have been fixed 2 days ago https://github.com/kiegroup/kie-tools/pull/1675/commits/ab2d38f95fb39141dd067653d2b6861f7355c30b
Can you confirm is working for you too? (the preview environment is updated)

>     * Is the only way how to get from settings to click the main logo? It might be confusing for the users to get back to main navigation.
Created issue: https://issues.redhat.com/browse/KOGITO-9257

**Comment by caponetto** [comment]
@fantonangeli https://github.com/kiegroup/kie-tools/pull/1683 renamed "KIE Sandbox Extended Services" entries to just "Extended Services". 
We now have some conflicts.

**Comment by tomasdavidorg** [comment]
@fantonangeli Thanks for the answers. I pull the latest changes.
- Confirming that the issue when models are selected and one of them is deleted is fixed. 
- About the quickstarts, I just wanted to make sure that this is intention. So it is ok as it is.

File: packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts (hunk 4 above)
Comment (fantonangeli) [reply packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts:20]: That is for the LocalStorage and doesn't clear the IndexedDB, as I can read.
As I could see on the web Cypress doesn't have any functionality to delete the indexedDB.

File: packages/serverless-logic-web-tools/src/AppConstants.ts (hunk 6 above)
Comment (caponetto) [reply packages/serverless-logic-web-tools/src/AppConstants.ts:19]: Thanks, Tomas! Here is the JIRA: https://issues.redhat.com/browse/KOGITO-9258

File: packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts (hunk 5 above)
Comment (fantonangeli) [reply packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts:47]: Behind the scenes, deleting the first row uses the indexedDB native implementation, same way as  per `deleteAllIndexedDBs`.
With `deleteAllIndexedDBs` I ensure that we start from a clear situation.
Please, let me know your viewpoint.

File: packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts (hunk 4 above)
Comment (tomasdavidorg) [reply packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts:20]: Thanks for the explanation.

File: packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts (hunk 5 above)
Comment (tomasdavidorg) [reply packages/serverless-logic-web-tools/it-tests/e2e/RecentModel.cy.ts:47]: Thanks. Makes sense to me. No issue in this case. 🙂

**Comment by fantonangeli** [comment]
@tomasdavidorg regarding:

> * There is no dialog after deleting a single line from the recent model table. Is this correct?

Because it requires more changes, I opened this jira for this activity:
https://issues.redhat.com/browse/KOGITO-9261

File: packages/serverless-logic-web-tools/src/settings/extendedServices/ExtendedServicesSettings.tsx [hunk 21]
```diff
@@ -0,0 +1,296 @@
...
+                    );
+                  }}
+                >
+                  Need help getting started? Follow our quickstart guide.
+                </Button>
+              </FormGroup>
+              <ActionGroup>
+                <Button
+                  isDisabled={!isCurrentConfigValid}
+                  id="kie-sandbox-extended-services-config-connect-button"
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/settings/extendedServices/ExtendedServicesSettings.tsx]:
```suggestion
                  id="extended-services-config-connect-button"
```

File: packages/serverless-logic-web-tools/src/navigation/Routes.ts [hunk 22]
```diff
@@ -131,6 +133,24 @@ export const routes = {
...
+  }>(({ workspaceId }) => `/${workspaceId}/files`),
+
+  recentModels: new Route<{}>(() => `/recent-models`),
+  sampleCatalog: new Route<{}>(() => `/sample-catalog`),
+
+  settings: {
+    home: new Route<{}>(() => SETTINGS_ROUTE),
+    github: new Route<{}>(() => `${SETTINGS_ROUTE}/github`),
+    openshift: new Route<{}>(() => `${SETTINGS_ROUTE}/openshift`),
+    extended_services: new Route<{}>(() => `${SETTINGS_ROUTE}/kie-sandbox-extended-services`),
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/navigation/Routes.ts]:
```suggestion
    extended_services: new Route<{}>(() => `${SETTINGS_ROUTE}/extended-services`),
```

**Review by caponetto — COMMENTED** [review]
Just a couple of suggestions. We are almost there :)

We still have some "KIE Sandbox Extended Services" entries in the quickstarts but I will work on them as part of [KOGITO-9258](https://issues.redhat.com/browse/KOGITO-9258)

**Review by caponetto — APPROVED** [review]

**Review by tomasdavidorg — APPROVED** [review]

**Pull request merged**